tests/test_indexes.py

```diff
@@ -0,0 +1,290 @@
+            "count of descriptors is not equal to the expected"
+        assert self.getCountOfDescriptorsInStorage(dynamicIndex) == dynamicIndex.bufSize, \
+            "wrong size of internal storage"
+
+    @staticmethod
+    def getCountOfDescriptorsInStorage(indexObject: Union[IndexBuilder, DynamicIndex, DenseIndex]) -> int:
+        """
+        Get actual count of descriptor in internal storage.
+        Args:
+            indexObject: an initialized index object with `getDescriptor` method
+        Returns:
+            count of descriptors
+        """
+        count = 0
+        while True:
+            try:
+                indexObject[count]
+            except IndexError:
+                break
+            else:
+                count += 1
+        return count
+
+    def test_build_empty_index(self):
+        """Test build empty index."""
+        expectedDescriptorsCount = 0
+        index = self.indexBuilder.buildIndex()
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        self.assertDynamicIndex(index, expectedDescriptorCount=expectedDescriptorsCount, expectedBufSize=0)
+
+    def test_append_descriptor_to_builder(self):
+        """Test append descriptor to index builder."""
+        expectedDescriptorsCount = 1
+        self.indexBuilder.append(self.faceDescriptor)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        assert expectedDescriptorsCount == self.getCountOfDescriptorsInStorage(self.indexBuilder)
+
+    def test_append_descriptors_batch_to_builder(self):
+        """Test append descriptors batch to index builder."""
+        expectedDescriptorsCount = 2
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        assert expectedDescriptorsCount == self.getCountOfDescriptorsInStorage(self.indexBuilder)
+
+    def test_get_descriptor_from_builder(self):
+        """Test get descriptor from internal storage."""
+        expectedDescriptorsCount = 2
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        for idx in range(expectedDescriptorsCount):
+            with self.subTest(case=f"get descriptor with index: {idx}"):
+                descriptor = self.indexBuilder[idx]
+                assert self.faceDescriptorBatch[idx].rawDescriptor == descriptor.rawDescriptor
+
+    def test_get_descriptor_from_builder_bad_index(self):
+        """Test get descriptor with invalid index from internal storage."""
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert len(self.faceDescriptorBatch) == self.indexBuilder.bufSize
+        nonexistentIndex = 2
+        with self.assertRaises(IndexError) as ex:
+            descriptor = self.indexBuilder[nonexistentIndex]
+        assert str(nonexistentIndex) in ex.exception.args[0]
```

Comment:
will this test be relevant after the fsdk fix?

Reply:
can we assert Exception?

Reply:
maybe we should leave this exception and not rely on fsdk? I have no idea how they will fix it :(